Features/organizationsectorindustry

mitchkirady opened the pull request:
PBI #537 and 510:
- Create option set for sector
- Add option set to Account entity
- Create industry entity
- Add M:M relationship between Account and Industry. Set relationship using PCF on form.
- Add M:M relationship between Complaint and Industry. Set relationship using PCF on form.
- Delete sector entity
- Remove sector from seed data

This does not include:
- Adding seed data for industries (requires PBI for data entry and adding to seed data)
- Filtering industry PCF (requires PBI for development of new PCF or improvements to current PCF)

mitchkirady commented on src/Compliance.Solution/Solution/Other/Customizations.xml:
```diff
@@ -19,6 +19,15 @@
   <AppModuleSiteMaps />
   <AppModules />
   <EntityDataProviders />
+  <connectionreferences>
+    <connectionreference connectionreferencelogicalname="opc_sharedcommondataserviceforapps_3e099">
+      <connectionreferencedisplayname>Common Data Service (current environment)</connectionreferencedisplayname>
+      <connectorid>/providers/Microsoft.PowerApps/apis/shared_commondataserviceforapps</connectorid>
+      <iscustomizable>1</iscustomizable>
+      <statecode>0</statecode>
+      <statuscode>1</statuscode>
+    </connectionreference>
+  </connectionreferences>
```
Connection References is a new feature (in preview) which includes canvas/flow connections in a solution. So this would be the reference for the "Common Data Service (current environment)" connection.

mitchkirady commented on src/Compliance.Solution/Solution/Workflows/Contact-CreateUpdate-Checkforduplicates-4D3E610E-F448-EA11-A812-000D3AF46CC5.json:
```diff
@@ -3,7 +3,9 @@
       "shared_commondataserviceforapps": {
         "runtimeSource": "embedded",
-        "connection": {},
+        "connection": {
+          "connectionReferenceLogicalName": "opc_sharedcommondataserviceforapps_3e099"
+        },
```
No changes were made to that flow, but MS would've added the connection reference in the json.

mitchkirady commented on src/Compliance.WebResources/src/services/ComplaintService.ts:
```diff
@@ -15,7 +15,7 @@ export class ComplaintService implements IComplaintService {
             .expand(x => x.opc_legislation, x => [x.opc_acronym])
             .expand(x => x.opc_opcpriorityid)
-            .expand(x => x.opc_sectorid)
+            //.expand(x => x.opc_sectorid)
```
Pierluc made this change, since I don't modify the TS. Looks like he decided to comment out the line instead, which seems fine to me as well for now, unless the team has agreed on specific coding standards.

mitchkirady commented on src/Compliance.WebResources/src/interfaces.ts:
```diff
@@ -94,7 +94,7 @@ export type TemplateEnvironmentVariable = {
 { opc_legislation: opc_legislation_Result } &
 { opc_opcpriorityid: opc_opcpriority_Result } &
-{ opc_sectorid: opc_sector_Result } &
+/*{ opc_sectorid: opc_sector_Result } &*/
```
Pierluc made this change, since I don't modify the TS. Looks like he decided to comment out the line instead, which seems fine to me as well for now, unless the team has agreed on specific coding standards.

mitchkirady commented on src/Compliance.Solution/Solution/Other/Solution.xml:
```diff
@@ -244,8 +246,6 @@
       <RootComponent type="92" id="{d7f77688-a5ac-ea11-a812-000d3af46fff}" behavior="0" />
       <RootComponent type="92" id="{e960b879-1816-ea11-a811-000d3af46489}" behavior="0" />
-      <RootComponent type="380" schemaName="opc_templatesapplication" behavior="0" />
-      <RootComponent type="381" id="{4710a2e7-53cc-ea11-a812-000d3af46489}" behavior="0" />
```
Thanks for the heads up, not sure how come these lines were removed, but it seems to happen often. I've added them to the next commit.

Sargatos reviewed (dismissed):
Looks all good to me.

mitchkirady commented:
- Added seed data for industries
- Added English and French Description fields on industry form
- Increased English and French Description field lengths to 7,000 due to seed data
